### PR TITLE
526 update upload mg

### PR DESCRIPTION
**Pull request by markgreenburg**
Updates the attachments schema to support new upload requirements

- Pulls attachment schema for private records and buddy statements from vets-json-schema
- Adds a new (well, it used to be there) 'Document type' dropdown to the uploader UI
- Functionally, changes attachments schema to `{ name, confirmationCode, attachmentId }`
  - `attachmentId` is actually the document type (from the VA-supplied upload types list)
  - `name` is the file name
  - `confirmationCode` is actually the GUID returned from `vets-api` upon successful document upload (used for matching form submissions to the right supporting document uploads)
- Pulls the attachments out of each disability and puts them into one array within `formData` for submission.

screen shot 2018-08-01 at 5 45 59 pm

screen shot 2018-08-01 at 5 46 24 pm

### CODE REVIEW

File: src/applications/disability-benefits/526EZ/config/form.js [hunk 1]
```diff
@@ -760,6 +763,13 @@ const formConfig = {
                         confirmationCode: response.data.attributes.guid
                       };
                     },
+                    // this is the uiSchema passed to FileField for the attachmentId schema
+                    // FileField requires this name be used
+                    attachmentSchema: {
+                      'ui:title': 'Document type'
+                    },
+                    // this is the uiSchema passed to FileField for the name schema
+                    // FileField requires this name be used
```
Comment (cvalarida) [inline src/applications/disability-benefits/526EZ/config/form.js]: :100:

File: src/applications/disability-benefits/526EZ/tests/config/documentUpload.unit.spec.jsx [hunk 2]
```diff
@@ -8,11 +8,13 @@ import { DefinitionTester, // selectCheckbox
 import formConfig from '../../config/form.js';
 import initialData from '../schema/initialData.js';
 
+// NOTE: This file probably duplicates recordUpload
```
Comment (cvalarida) [inline src/applications/disability-benefits/526EZ/tests/config/documentUpload.unit.spec.jsx]: Looks like they're two separate pages:
- `documentUpload` is for lay statements and other uploads
- `recordUpload` is for private medical records

I definitely had to look that up, though. Could probably use a rename.

**Review by cvalarida — APPROVED** [review]

File: src/applications/disability-benefits/526EZ/tests/config/documentUpload.unit.spec.jsx (hunk 2 above)
Comment (markgreenburg) [reply src/applications/disability-benefits/526EZ/tests/config/documentUpload.unit.spec.jsx]: Yeah, I finally came to the same conclusion as well. I think I'll rename them in a separate PR.

File: src/applications/disability-benefits/526EZ/config/form.js (hunk 1 above)
Comment (markgreenburg) [reply src/applications/disability-benefits/526EZ/config/form.js]: 🤷‍♂️

**Pull request merged**